Failed to build with cmake >= 4.0

<!-- This is a suggested template for your issue. (remove as you check)     -->
<!--   * After crash, always backup and include output.txt in the bug report-->
<!--     from your pioneer config folder, before starting a new game        -->
<!--     (consult FAQ on wiki for info on how to find it).                  -->
<!--   * if graphical: also include opengl.txt from pioneer config dir      -->
<!--   * Please Include version of your last bug-free Pioneer, if possible. -->
<!--   * Has the issue been reported before? Search the issue tracker.      -->
<!--   * Perhaps a screenshot illustrates the bug well?                     -->

Due to the [obsolescence of versions over 3.5](https://cmake.org/cmake/help/latest/release/4.0.html#deprecated-and-removed-features), some of the sub-projects give an error at the generation stage:
```
CMake Error at contrib/nanosockets/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 has been removed from CMake.

CMake Error at contrib/lz4/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 has been removed from CMake.
```

I think this soon can cause some unnecessary problems for those who will build a release from the source code. Can we simply increase the versions in the corresponding CMakeLists.txt?

__My pioneer version (and OS):__
master at 003d8c0 (25-04-28)

--- a/contrib/lz4/CMakeLists.txt
+++ b/contrib/lz4/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.4)
+cmake_minimum_required(VERSION 3.5)
 project(lz4 LANGUAGES C)
 
 list(APPEND LZ4_SOURCES

--- a/contrib/nanosockets/CMakeLists.txt
+++ b/contrib/nanosockets/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.1)
+cmake_minimum_required(VERSION 3.5)
 project(nanosockets C)
 
 if (MSYS OR MINGW)

